File: BEmu/IntradayBarRequest/IntradayBarRequestElementBool.cs
//------------------------------------------------------------------------------
// <copyright project="BEmu_csh" file="IntradayBarRequest/IntradayBarRequestElementBool.cs" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

namespace Bloomberglp.Blpapi.IntradayBarRequest
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    internal class IntradayBarRequestElementBool : IntradayBarRequestElementString
    {
        private readonly bool _value;

        internal IntradayBarRequestElementBool(string elementName, bool value)
            : base(elementName, value.ToString().ToLower())
        {
            this._value = value;
        }

        //I can't override GetElementAsBool here because the Bloomberg Request object stores bools as strings, not bools.  You can't convert the string to a bool
        internal bool GetBool { get { return this._value; } }
        public override Schema.Datatype Datatype { get { return Schema.Datatype.BOOL; } }
    }
}

File: BEmu_cpp/src/IntradayBarRequest/IntradayBarRequestElementBool.cpp
//------------------------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/IntradayBarRequest/IntradayBarRequestElementBool.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------------------------

#include "BloombergTypes/Name.h"
#include "BloombergTypes/ElementPtr.h"
#include "IntradayBarRequest/IntradayBarRequestElementBool.h"

namespace BEmu
{
	namespace IntradayBarRequest
	{
		IntradayBarRequestElementBool::IntradayBarRequestElementBool(const std::string& elementName, bool bvalue) : CanConvertToStringType(bvalue, false)
		{
			this->_elementName = elementName;
			this->_bvalue = bvalue;
		}

		IntradayBarRequestElementBool::~IntradayBarRequestElementBool()
		{
		}

		bool IntradayBarRequestElementBool::getBool() const
		{
			return this->_bvalue;
		}

		Name IntradayBarRequestElementBool::name() const
		{
			Name result(this->_elementName.c_str());
			return result;
		}

		size_t IntradayBarRequestElementBool::numValues() const
		{
			return 1;
		}

		size_t IntradayBarRequestElementBool::numElements() const
		{
			return 0;
		}

		bool IntradayBarRequestElementBool::isArray() const
		{
			return false;
		}

		bool IntradayBarRequestElementBool::isComplexType() const
		{
			return false;
		}

		const char * IntradayBarRequestElementBool::getValueAsString(int index) const
		{
			return this->ValueAsString().c_str();
		}

		std::ostream& IntradayBarRequestElementBool::print(std::ostream& stream, int level, int spacesPerLevel) const
		{
			this->prettyPrintHelper(stream, level, spacesPerLevel, this->ValueAsString());
			return stream;
		}

	}
}